### PR TITLE
done

### PR DESCRIPTION
**Pull request by Richaa2**

### CODE REVIEW

File: src/main/java/core/mate/academy/service/BulldozerProducer.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package core.mate.academy.service;
+
+import core.mate.academy.model.Bulldozer;
+import java.util.ArrayList;
+import java.util.List;
+
+public class BulldozerProducer implements MachineProducer<Bulldozer> {
+    private List<Bulldozer> bulldozerList = new ArrayList<>();
```
Comment (aleksandr-bevza) [inline src/main/java/core/mate/academy/service/BulldozerProducer.java:8]:
```suggestion
```

File: src/main/java/core/mate/academy/service/BulldozerProducer.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+    @Override
+    public List<Bulldozer> get() {
```
Comment (aleksandr-bevza) [inline src/main/java/core/mate/academy/service/BulldozerProducer.java:11]: You should return new List of buldozers every time this method is invoked. Don't use "bulldozerList" as class field.
Also fix it in other Producers

File: src/main/java/core/mate/academy/service/ExcavatorProducer.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package core.mate.academy.service;
+
+import core.mate.academy.model.Excavator;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ExcavatorProducer implements MachineProducer<Excavator> {
+    private List<Excavator> bulldozerList = new ArrayList<>();
```
Comment (aleksandr-bevza) [inline src/main/java/core/mate/academy/service/ExcavatorProducer.java:8]: Fix it like in BulldozerProducer class

File: src/main/java/core/mate/academy/service/TruckProducer.java [hunk 4]
```diff
@@ -0,0 +1,15 @@
+package core.mate.academy.service;
+
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
+public class TruckProducer implements MachineProducer<Truck> {
+    private List<Truck> bulldozerList = new ArrayList<>();
```
Comment (aleksandr-bevza) [inline src/main/java/core/mate/academy/service/TruckProducer.java:8]: Fix it like in BulldozerProducer class

File: src/main/java/core/mate/academy/service/MachineService.java [hunk 5]
```diff
@@ -16,7 +16,8 @@ public interface MachineService {
      * @param type - any class of Machine sub class. For example: Truck.class or Bulldozer.class
      * @return the list of machines
      */
-    List<Object> getAll(Class type);
+    List<T> getAll(Object type);
```
Comment (aleksandr-bevza) [inline src/main/java/core/mate/academy/service/MachineService.java:19]:
```suggestion
    List<T> getAll(Class<? extends T> type);
```

File: src/main/java/core/mate/academy/service/MachineServiceImpl.java [hunk 6]
```diff
@@ -1,7 +1,38 @@
...
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl<T extends Machine> implements MachineService<T> {
+    @Override
+    public List<T> getAll(Object type) {
```
Comment (aleksandr-bevza) [inline src/main/java/core/mate/academy/service/MachineServiceImpl.java:15]:
```suggestion
    public List<T> getAll(Class<? extends Machine> type) {
```

File: src/main/java/core/mate/academy/service/MachineServiceImpl.java [hunk 7]
```diff
@@ -1,7 +1,38 @@
...
+
```
Comment (aleksandr-bevza) [inline src/main/java/core/mate/academy/service/MachineServiceImpl.java:16]: remove empty line

File: src/main/java/core/mate/academy/service/MachineServiceImpl.java [hunk 8]
```diff
@@ -1,7 +1,38 @@
...
+        if (type == Bulldozer.class) {
+            return (List<T>) new BulldozerProducer().get();
```
Comment (aleksandr-bevza) [inline src/main/java/core/mate/academy/service/MachineServiceImpl.java:18]: Create your producers as class fields. Don't create new Producer every time you call "getAll" method.
Don't forget about abstract reference type MachineProducer

**Review by aleksandr-bevza — CHANGES_REQUESTED** [review]
In this task you need to populate Bulldozer, Excavator and Truck classes with some type-specific fields. Add them please

**Review by zachar-chorny — APPROVED** [review]

**Pull request closed without merging**